### PR TITLE
Remove all compile repitions for multifile input

### PR DESCRIPTION
**Pull request by JorikSchellekens**
We remove all redundancies in compile by deferring to solc. Solc produces
a single linked ast for multiple file inputs so this resolves the full dependecy
chain.

There is a remarkable speed up to UniStark as a result of this pr.

### CODE REVIEW

File: src/solCompile.ts [hunk 1]
```diff
@@ -60,14 +78,16 @@ type SolcInput = {
   };
 };
 
-function formatInput(fileName: string): SolcInput {
+function formatInput(fileNames: string[]): SolcInput {
+  const sources: { [key: string]: { urls: string[] } } = {};
```
Comment (swapnilraj) [inline src/solCompile.ts:82]: minor nit: let's make this a reduce

**Comment by swapnilraj** [comment]
does hardhat-warp; do we invoke warp somewhat like this: `warp transpile <all contracts> --compile-cairo`

File: src/solCompile.ts [hunk 2]
```diff
@@ -17,19 +17,37 @@ import { error } from './utils/formatting';
...
+function compileSolFilesCommon(
+  files: string[],
+): { result: unknown; compilerVersion: string } {
+  const sources = files.map((file) => {
+    return getSolFileVersion(file);
+  });
+
+  sources.forEach((version, i) => {
+    const [, majorVersion] = matchCompilerVersion(version);
+    if (majorVersion != '7' && majorVersion != '8') {
```
[the remark this answers is not in the document]
Comment (JorikSchellekens) [reply src/solCompile.ts:29]: I haven't changed it from what it was. I presume we won't support it for much longer @GuyMcComb @swapnilraj ?

File: src/solCompile.ts (hunk 1 above)
Comment (JorikSchellekens) [reply src/solCompile.ts:82]: This will just make it longer since we'll still need to declare 'sources' to get the types to work. I already tried it.

Comment (JorikSchellekens) [reply src/solCompile.ts:82]:
```
  const sources = fileNames.reduce((sources, fileName) => {
    sources[fileName] = {
      urls: [fileName],
    };
    return sources;
  }, {} as { [key: string]: { urls: string[] } });
  ```

File: src/solCompile.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (swapnilraj) [reply src/solCompile.ts:27]: I am in support of dropping solidity 0.7 support

[the remark this answers is not in the document]
Comment (JorikSchellekens) [reply src/solCompile.ts:27]: Me too but let's leave it to a seperate pr, we need to make sure the messaging is consistent across docs

**Comment by swapnilraj** [comment]
@GuyMcComb can you start off semantic tests for this PR

**Review by swapnilraj — APPROVED** [review]

**Pull request merged**